clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroComputeRequestBuilder.java

```diff
@@ -256,24 +248,31 @@ protected List<Schema.Field> getCommonResultFields() {
        *    value in Java format, but the {@link Schema.Field#equals} will compare the underlying {@link com.fasterxml.jackson.databind.JsonNode}
        *    format of the default value.
        */
-      resultSchemaFields.add(AvroCompatibilityHelper.newField(existingField).setDoc("").build());
+      if (existingField != null) {
+        resultSchemaFields.add(AvroCompatibilityHelper.newField(existingField).setDoc("").build());
+      }
     });
     dotProducts.forEach(dotProduct -> {
       Schema.Field dotProductField = AvroCompatibilityHelper
-          .createSchemaField(dotProduct.resultFieldName.toString(), getDotProductResultSchema(), "", null);
+          .createSchemaField(dotProduct.resultFieldName.toString(), DOT_PRODUCT_RESULT_SCHEMA, "", null);
       resultSchemaFields.add(dotProductField);
     });
     cosineSimilarities.forEach(cosineSimilarity -> {
       Schema.Field cosineSimilarityField = AvroCompatibilityHelper
-          .createSchemaField(cosineSimilarity.resultFieldName.toString(), getCosineSimilarityResultSchema(), "", null);
+          .createSchemaField(cosineSimilarity.resultFieldName.toString(), COSINE_SIMILARITY_RESULT_SCHEMA, "", null);
       resultSchemaFields.add(cosineSimilarityField);
     });
     hadamardProducts.forEach(hadamardProduct -> {
       Schema.Field hadamardProductField = AvroCompatibilityHelper
           .createSchemaField(hadamardProduct.resultFieldName.toString(), HADAMARD_PRODUCT_RESULT_SCHEMA, "", null);
       resultSchemaFields.add(hadamardProductField);
     });
-    resultSchemaFields.add(VENICE_COMPUTATION_ERROR_MAP_FIELD);
+    resultSchemaFields.add(
+        AvroCompatibilityHelper.createSchemaField(
```

Comment:
Can you copy over the original javadoc for this `error map` prevent someone from refactoring this to be a static field in the future?

Reply:
Sure